barretenberg/cpp/src/barretenberg/dsl/acir_format/witness_constant.cpp

```diff
@@ -23,18 +23,11 @@ bb::stdlib::cycle_group<Builder> to_grumpkin_point(const WitnessOrConstant<FF>&
     auto point_y = to_field_ct(input_y, builder);
     auto infinite = bool_ct(to_field_ct(input_infinite, builder));
 
-    // When we do not have the witness assignments, we set is_infinite value to true if it is not constant
-    // else default values would give a point which is not on the curve and this will fail verification
+    // If no witness, set coordinates to correspond to an arbitrary valid curve point (affine_one)
     if (!has_valid_witness_assignments) {
-        if (!input_infinite.is_constant) {
```

Comment:
This branch was leading to the on_curve assertion failing in the cycle group constructor due to an odd quirk in `bool_t`. In particular, we define the `get_value()` on `bool_t` as `bool get_value() const { return witness_bool ^ witness_inverted; }`, i.e. we do not actually utilize the underlying witness value stored in `variables` which is all that has been updated in this conditional branch. It might be possible to fix this but I've opted to simply remove this branch since it does not seem to be necessary. By definition we need it to be the case that our circuit logic does not depend on the underlying witness value for `is_infinite`.